src/IW/App/Error.hs

```diff
@@ -118,15 +123,22 @@ data AppErrorType
     | UrlDownloadFailed Url
     deriving (Show, Eq)
 
--- | Map the @github@ library's @Error@ type into AppErrorType.
+newtype CabalPError = CabalPError { unCabalPError :: PError }
+    deriving (Show)
+
+instance Eq CabalPError where
+    (CabalPError (PError pos1 str1)) == (CabalPError (PError pos2 str2)) =
+        pos1 == pos2 && str1 == str2
+
+-- | Map the @github@ library's @Error@ type into @AppErrorType@.
 githubErrToAppErr :: GitHub.Error -> AppErrorType
 githubErrToAppErr = \case
     GitHub.HTTPError e    -> GithubHttpError $ show e
     GitHub.ParseError msg -> GithubParseError msg
     GitHub.JsonError msg  -> GithubJsonError msg
     GitHub.UserError msg  -> GithubUserError msg
 
--- | Map 'AppError' into a HTTP error code.
+-- | Map @AppError@ into a HTTP error code.
```

Comment:
I guess `AppError` is the correct reference here. Any reason, you don't want to link to the data type in docs?

Reply:
@vrom911 I'm gonna change this back. I'm just finding out what `'...'` does. My mistake I need to learn the haddock syntax because this is a super useful tool. Anyways, should I also link to functions as well? I use `@...@` a lot, but I think I'm going to replace a lot of them. Before the final submit I'm parsing through each module and making sure the documentation is up to date, so let me know if this is a good idea. In what cases should `@...@` be used? It seems to me that you would always want to link to the function/data type docs.

Reply:
Yes, your assumption is correct, for all the functions, data types etc. you would want to make a link to it, that is super useful! `@...@` is used for code blocks, for example, you may want to write @2 + 2@, or else.